app/src/services/coms.ts

```diff
@@ -22,28 +23,52 @@ function comsAxios(options: AxiosRequestConfig = {}): AxiosInstance {
 }
 
 const service = {
+  /**
+   * @function createBucket
+   * Creates a bucket record. Bucket should exist in S3. If the set of bucket, endpoint and key match
+   * an existing record, the user will be added to that existing bucket with the provided permissions
+   * instead of generating a new bucket record.
+   * This endpoint can be used to grant the current user permission to upload to a new or existing bucket.
+   * @param {string | null | undefined} bearerToken The bearer token of the authorized user
+   * @param {Action[]} permissions An array of permissions to grant the user
+   */
+  async createBucket(bearerToken: string | null | undefined, permissions: Array<Action>) {
+    const { data } = await comsAxios({
+      headers: { Authorization: `Bearer ${bearerToken}` }
+    }).put('/bucket', {
+      accessKeyId: config.get('server.objectStorage.accessKeyId'),
+      bucket: config.get('server.objectStorage.bucket'),
+      bucketName: 'PCNS',
+      endpoint: config.get('server.objectStorage.endpoint'),
+      secretAccessKey: config.get('server.objectStorage.secretAccessKey'),
+      key: config.get('server.objectStorage.key'),
+      permCodes: permissions
+    });
+    return data;
+  },
+
   /**
    * @function getObject
    * Get an object
-   * @param {IncomingHttpHeaders} incomingHeaders The request headers
+   * @param {string | null | undefined} bearerToken The bearer token of the authorized user
    * @param {string} objectId The id for the object to get
    */
-  async getObject(incomingHeaders: IncomingHttpHeaders, objectId: string) {
+  async getObject(bearerToken: string | null | undefined, objectId: string) {
     const { status, headers, data } = await comsAxios({
       responseType: 'arraybuffer',
-      headers: { Authorization: incomingHeaders.authorization }
+      headers: { Authorization: `Bearer ${bearerToken}` }
     }).get(`/object/${objectId}`);
     return { status, headers, data };
   },
 
   /**
    * @function getObjects
    * Gets a list of objects
-   * @param {IncomingHttpHeaders} incomingHeaders The request headers
+   * @param {string | null | undefined} bearerToken The bearer token of the authorized user
    * @param {string[]} objectIds Array of object ids to get
    */
-  async getObjects(incomingHeaders: IncomingHttpHeaders, objectIds: Array<string>) {
-    const { data } = await comsAxios({ headers: { Authorization: incomingHeaders.authorization } }).get('/object', {
+  async getObjects(bearerToken: string | null | undefined, objectIds: Array<string>) {
+    const { data } = await comsAxios({ headers: { Authorization: `Bearer ${bearerToken}` } }).get('/object', {
```

Comment:
Safety: There should be a conditional if here to drop the Authorization header completely if `bearerToken` is either null or undefined.